common/doc-dev/3_How_to_set_up_openssh_server_for_ssh_unit_tests.md

```diff
@@ -0,0 +1,227 @@
+# How to set up a local `openssh-server` to enable ssh unit tests
+
+# Motivation
+
+`ssh`-based unit tests are skipped in `make test` when no local ssh server
+is configured and running.
+
+In order to execute also run all ssh unit tests in the `common` folder
+via `make test` the `openssh` server must be **installed on your local machine**
+and a public/private key must be set up for password-less connections.
+
+This document describes the required steps.
+
+# How the unit tests access the ssh server
+
+`ssh`-based unit tests use the [`common/test/generic.py`](https://github.com/bit-team/backintime/blob/f801b14a98f9a442008a5f514eec98e1b2d7e29a/common/test/generic.py)
+to check and establish a ssh connection to the ssh server via this code:
+
+https://github.com/bit-team/backintime/blob/f801b14a98f9a442008a5f514eec98e1b2d7e29a/common/test/generic.py#L43-L72
+
+The code implements the following logic:
+
+1. Check if a `sshd` process is running on the local machine
+2. Check if a private key file `~/.ssh/id_rsa` exists
+3. Check if the file `~/.ssh/authorized_keys` exists
+4. Check if the private key from `id_rsa` is contained in `authorized_keys`
```

Comment:
Only public keys should be in `authorized_keys`

Reply:
I never do such detailed checks when creating setup passwordless SSH login. But if this needs a correction it could be go into my open PR #1561 .

Reply:
I think a direct edit without a PR should suffice to fix my stupid mistake :smiley: 
Shall I do this?

Reply:
Yep.

Reply:
All right, fixed with https://github.com/bit-team/backintime/commit/65d0585b056f4af299e0d87eeb08e77c202457c6

https://github.com/bit-team/backintime/blob/dev/common/doc-dev/3_How_to_set_up_openssh_server_for_ssh_unit_tests.md#how-the-unit-tests-access-the-ssh-server

THX for the review (@buhtz and the nice code refactoring of `generic.py` public key check algorithm in Sept 2023 which us much easier to understand now)